src/core/IO/Socket/Async.pm6

```diff
@@ -242,42 +267,43 @@ my class IO::Socket::Async {
-                                nqp::bindattr($client_socket, IO::Socket::Async,
+                                my \client := nqp::create(IO::Socket::Async);
+                                nqp::bindattr(client, IO::Socket::Async,
                                     '$!VMIO', socket);
-                                nqp::bindattr($client_socket, IO::Socket::Async,
+                                nqp::bindattr(client, IO::Socket::Async,
                                     '$!enc', $!encoding.name);
-                                nqp::bindattr($client_socket, IO::Socket::Async,
+                                nqp::bindattr(client, IO::Socket::Async,
                                     '$!encoder', $!encoding.encoder());
-                                nqp::bindattr($client_socket, IO::Socket::Async,
+                                nqp::bindattr(client, IO::Socket::Async,
                                     '$!peer-host', peer-host);
-                                nqp::bindattr($client_socket, IO::Socket::Async,
+                                nqp::bindattr(client, IO::Socket::Async,
                                     '$!peer-port', peer-port);
-                                nqp::bindattr($client_socket, IO::Socket::Async,
+                                nqp::bindattr(client, IO::Socket::Async,
                                     '$!socket-host', socket-host);
-                                nqp::bindattr($client_socket, IO::Socket::Async,
+                                nqp::bindattr(client, IO::Socket::Async,
                                     '$!socket-port', socket-port);
-                                setup-close($client_socket);
-                                emit($client_socket);
+                                setup-close(client);
+                                emit(client);
                             }
                             elsif socket-host {
                                 $host-vow.keep(~socket-host);
                                 $port-vow.keep(+socket-port);
                             }
                         }
                     },
-                    $!host, $!port, $!backlog, SocketCancellation);
+                    nqp::decont($VMIO), $!host, $!port, $!backlog, SocketCancellation);
                 $tap = ListenSocket.new: {
                     my $p = Promise.new;
                     my $v = $p.vow;
                     nqp::cancelnotify($cancellation, $!scheduler.queue, { $v.keep(True); });
                     $p
-                }, :$socket-host, :$socket-port;
+                }, :$VMIO, :$socket-host, :$socket-port;
                 tap($tap);
+
                 CATCH {
                     default {
-                        tap($tap = ListenSocket.new({ Nil },
-                            :$socket-host, :$socket-port)) unless $tap;
+                        tap(ListenSocket.new: { Nil }, :$VMIO,
```

Comment:
This is still not equivalent to the previous code, which assigned the `ListenSocket` to `$tap`

Reply:
Oop